chain/indexer/registry.go

```diff
@@ -0,0 +1,101 @@
+package indexer
+
+const (
+	// blocks
+	BlocksTask = "blocks"
+
+	// consensus
+	ChainConsensusTask = "chain_consensus"
```

Comment:
Why do we use both `snake_case` and `nocase` for task name representations. I assume (hope) the snake_case variant is never exposed publicly?

Reply:
We only use `snake_case`, these strings, eg. `chain_consensus` are provided by the used on the command line. Would you prefer `nocase`?

Reply:
The inconsistency in the interface from the prior version rubs me the wrong way (interface changes are always painful for the user). While there's nothing wrong with changing it, I would personally err on the side of leaving things as consistent as possible from one version to the next unless there's a good reason.

On the other hand, with the interface changing so drastically here I can be less bothered by the new arg pattern. And the division of existing tasks does make its sub-tables/task (which are now tasks themselves) somewhat awkward to name without changing all of them.

Arguing for/against doesn't leave me with strong feels one way or the other, so please decide what you think is best here and I'll support downstream. #bikeshedding 

If you leave it and in case you haven't decided/considered this, I would at least want next release that includes this refactor to have a minor version bump rather than a revision bump. 🙏

Reply:
The change here is that we no longer have tasks that produce many models, rather there is (mostly) a one-to-one mapping between the task name and the model it produces. There isn't really a new arg pattern here, rather there a different args altogether.